### PR TITLE
Add auto_paginate to options

### PR DESCRIPTION
**Pull request by StoneFrog**
To be merged after #69

### CODE REVIEW

**Comment by Fredar** [comment]
Whats the usecase of forcing auto_paginate to false?

File: lib/synced/strategies/full.rb [hunk 1]
```diff
@@ -183,7 +184,7 @@ def api_request_options
             options[:include] += @include
           end
           options[:fields] = @fields if @fields.present?
-          options[:auto_paginate] = true
+          options[:auto_paginate] = true if @auto_paginate
```
Comment (Fredar) [inline lib/synced/strategies/full.rb]: `options[:auto_paginate] = @auto_paginate` ?

Comment (LeonidMorozov) [inline lib/synced/strategies/full.rb]: `options[:auto_paginate] = !!@auto_paginate` ?

**Comment by StoneFrog** [comment]
@Fredar 
There is another PR to make it possible to switch between using auto_paginate (so fetching and saving  everything at once) and pagination with blocks (saving batch after each request). As Manda didn't want all of that mixed up in one PR I extracted it here.
So to provide backwards compatibility, but at the same time to allow quick switching between auto_paginate and pagination with blocks for testing, I decided to add this option.
Also since before merging this other PR passing `auto_paginate: false` would probably break something, I didn't update any docs here.

If it's a bit too confusing I may add this option after this other PR is merged.

File: lib/synced/strategies/full.rb (hunk 1 above)
Comment (StoneFrog) [inline lib/synced/strategies/full.rb]: You're right, when writing I already had in mind that it is just temporary solution and will be changed shortly. Wouldn't hurt to correct it though.

**Comment by Mandaryn** [comment]
this is going to be configurable shortly with block pagination, extracted here to make the other pr smaller

**Pull request merged**